### PR TITLE
Update CIM lib

### PR DESCRIPTION
**Pull request by rawagner**
This PR also includes changes from @jeswanke https://github.com/open-cluster-management/console/pull/826

Additional permissions to watch BMH resource is also needed https://github.com/open-cluster-management/console-chart/pull/153

Signed-off-by: Rastislav Wagner <rawagner@redhat.com>

### CODE REVIEW

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx [hunk 1]
```diff
@@ -36,7 +42,9 @@ import { ProviderConnection, unpackProviderConnection } from '../../../../../res
 import { Secret } from '../../../../../resources'
 import { createResource as createResourceTool } from '../../../../../resources'
 import { FeatureGates } from '../../../../../FeatureGates'
-import { getNetworkingPatches } from '../components/cim/utils'
+import { getNetworkingPatches } from './components/assisted-installer/utils'
+import { NetworkConfigurationValues } from 'openshift-assisted-ui-lib/dist/src/common/types/clusters'
```
Comment (mareklibra) [inline frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx:46]: We should use imports via the `CIM` namepsace. If needed, please re-export in the ai-ui-lib

Comment (rawagner) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx:46]: we can do that in next release of ai-ui-lib

File: frontend/tsconfig.json [hunk 2]
```diff
@@ -32,7 +32,8 @@
         "moduleResolution": "node",
         "resolveJsonModule": true,
         "isolatedModules": true,
-        "noEmit": true
+        "noEmit": true,
+        "useUnknownInCatchVariables": false,
```
Comment (jamestalton) [inline frontend/tsconfig.json:36]: I would like to avoid turning useUnknownInCatchVariables off

**Review by jamestalton — CHANGES_REQUESTED** [review]

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx (hunk 1 above)
Comment (rawagner) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/CreateCluster.tsx]: fixed

File: frontend/tsconfig.json (hunk 2 above)
Comment (rawagner) [reply frontend/tsconfig.json]: ok, I removed this change

**Review by jamestalton — APPROVED** [review]
lgtm

**Pull request merged**